unitTests/meson.build

```diff
@@ -0,0 +1,56 @@
+gtest_dep = dependency('gtest_main', required: get_option('unitTests'))
+if not gtest_dep.found()
+  subdir_done()
+endif
+
+gmock_dep = dependency('gmock')
+
+test_sources = [
+  'mainTestRunner.cpp',
+  'test_basicio.cpp',
+  'test_bmpimage.cpp',
+  'test_cr2header_int.cpp',
+  'test_datasets.cpp',
+  'test_Error.cpp',
+  'test_DateValue.cpp',
+  'test_enforce.cpp',
+  'test_FileIo.cpp',
+  'test_futils.cpp',
+  'test_helper_functions.cpp',
+  'test_image_int.cpp',
+  'test_ImageFactory.cpp',
+  'test_jp2image.cpp',
+  'test_jp2image_int.cpp',
+  'test_IptcKey.cpp',
+  'test_LangAltValueRead.cpp',
+  'test_Photoshop.cpp',
+  'test_pngimage.cpp',
+  'test_safe_op.cpp',
+  'test_slice.cpp',
+  'test_tiffheader.cpp',
+  'test_types.cpp',
+  'test_TimeValue.cpp',
+  'test_utils.cpp',
+  'test_XmpKey.cpp',
+]
+
+if get_option('video')
+  test_sources += [
+    'test_asfvideo.cpp',
+    'test_matroskavideo.cpp',
+    'test_riffVideo.cpp',
+  ]
+endif
+
+t_args = ['-DTESTDATA_PATH="@0@"'.format('..' / 'test' / 'data')]
+
+privinc = include_directories('../src', '../include/exiv2')
+unit_tests = executable(
+  test_sources,
```

Comment:
Missing executable name. This executable is therefore `unitTests/mainTestRunner.cpp.exe` and it doesn't include the main() implementation from there (instead it uses gtest_main?).

Reply:
knew there was some reason files() didn't work. Thanks.

Reply:
yup, in Meson git master the python traceback is fixed with real type checking:
```
unitTests/meson.build:48:0: ERROR: executable argument 1 was of type "File" but should have been "str"
```